File: src/backend/app/schemas/escalation.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel
from app.models.escalation import EscalationStatus, EscalationPriority


class EscalationCreateRequest(BaseModel):
    conversation_id: str
    reason: str
    user_notes: Optional[str] = None
    priority: EscalationPriority = EscalationPriority.MEDIA


class EscalationResolveRequest(BaseModel):
    resolution_notes: str
    status: EscalationStatus = EscalationStatus.RESOLVIDO


class EscalationResponse(BaseModel):
    id: int
    conversation_id: str
    student_id: int
    student_name: Optional[str] = None
    student_ra: Optional[str] = None
    assigned_attendant_id: Optional[int] = None
    assigned_attendant_name: Optional[str] = None
    reason: str
    user_notes: Optional[str] = None
    priority: EscalationPriority
    status: EscalationStatus
    resolution_notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    resolved_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: src/mobile/lib/models/escalation.dart
class EscalationModel {
  final int id;
  final String conversationId;
  final int studentId;
  final String? studentName;
  final String? studentRa;
  final int? assignedAttendantId;
  final String? assignedAttendantName;
  final String reason;
  final String? userNotes;
  final String priority;
  final String status;
  final String? resolutionNotes;
  final DateTime createdAt;
  final DateTime? resolvedAt;

  EscalationModel({
    required this.id,
    required this.conversationId,
    required this.studentId,
    this.studentName,
    this.studentRa,
    this.assignedAttendantId,
    this.assignedAttendantName,
    required this.reason,
    this.userNotes,
    required this.priority,
    required this.status,
    this.resolutionNotes,
    required this.createdAt,
    this.resolvedAt,
  });

  factory EscalationModel.fromJson(Map<String, dynamic> json) {
    return EscalationModel(
      id: json['id'] ?? 0,
      conversationId: json['conversation_id'] ?? '',
      studentId: json['student_id'] ?? 0,
      studentName: json['student_name'],
      studentRa: json['student_ra'],
      assignedAttendantId: json['assigned_attendant_id'],
      assignedAttendantName: json['assigned_attendant_name'],
      reason: json['reason'] ?? '',
      userNotes: json['user_notes'],
      priority: json['priority'] ?? 'MEDIA',
      status: json['status'] ?? 'PENDENTE',
      resolutionNotes: json['resolution_notes'],
      createdAt: json['created_at'] != null ? DateTime.parse(json['created_at']) : DateTime.now(),
      resolvedAt: json['resolved_at'] != null ? DateTime.parse(json['resolved_at']) : null,
    );
  }
}
